src/lv_objx/lv_spinbox.c

```diff
@@ -381,17 +381,20 @@ static void lv_spinbox_updatevalue(lv_obj_t * spinbox)
 
     char buf[LV_SPINBOX_MAX_DIGIT_COUNT + 8];
     memset(buf, 0, sizeof(buf));
-    char * buf_p = buf;
-
-    /*Add the sign*/
-    (*buf_p) = ext->value >= 0 ? '+' : '-';
-    buf_p++;
-
+    char* buf_p = buf;
     int i;
-    /*padding left*/
-    for(i = 0; i < ext->digit_padding_left; i++) {
-        (*buf_p) = ' ';
+    
+    if (ext->range_min < 0) { // hide sign if there are only positive values
+       
+        /*Add the sign*/
+        (*buf_p) = ext->value >= 0 ? '+' : '-';
         buf_p++;
+
+        /*padding left*/
+        for (i = 0; i < ext->digit_padding_left; i++) {
```

Comment:
Padding should be applied independently from the sign.
So this `for` should be out of `if (ext->range_min < 0) {`

Reply:
you are right! I've corrected code.